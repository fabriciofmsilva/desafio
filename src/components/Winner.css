.Winner {
  max-width: 520px;
  height: 52px;
  margin: 2rem auto;
  padding: 1rem;
  font-size: 2rem;
  font-weight: bold;
  color: #27ae60;
  background-color: #ecf0f1;
  visibility: visible;
  transition: all .5s ease-in-out;
}

.Winner.hidden {
  display: none;
}

.Winner.zoomIn {
  animation: zoomIn .8s;
  animation-fill-mode: both;
}

@keyframes zoomIn {
  from {
    opacity: 0;
    transform: scale3d(.3, .3, .3);
  }

  50% {
    opacity: 1;
  }
}
