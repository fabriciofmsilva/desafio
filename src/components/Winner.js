import React from 'react';
import { connect } from 'react-redux';

import { reset } from '../actions/game';

import './Winner.css';

export class Winner extends React.Component {

  getWinner = () => {
    return this.props.players.filter((player) => {
      return player.life > 0;
    });
  };

  render() {
    return (
      <div className={`Winner ${this.props.gameover ? 'zoomIn' : 'hidden'}`}>
        {this.getWinner()[0].name} venceu!
      </div>
    );
  }
}

const mapStateToProps = (state, props) => ({
  gameover: state.game.over,
  players: state.game.players
});

export default connect(mapStateToProps)(Winner);
