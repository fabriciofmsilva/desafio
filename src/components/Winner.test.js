import React from 'react';
import { shallow } from 'enzyme';

import { Winner } from './Winner';
import gameDefaultState from '../test/fixtures/game';

let wrapper;

beforeEach(() => {
  wrapper = shallow(<Winner players={gameDefaultState.players} gameover={false} />);
});

test('should render Winner correctly when gameover is false', () => {

  expect(wrapper).toMatchSnapshot();
  expect(wrapper.contains('Jogador 1')).toEqual(true);
});

test('should render Winner correctly when gameover is true', () => {
  wrapper.setProps({ gameover: true });

  expect(wrapper).toMatchSnapshot();
  expect(wrapper.contains('Jogador 1')).toEqual(true);
});

test('should render Winner correctly', () => {
  wrapper.setProps({ players: [
    {
      id: 1,
      name: 'Jogador 1',
      life: 0
    },
    {
      id: 2,
      name: 'Jogador 2',
      life: 100
    }
  ]});

  expect(wrapper).toMatchSnapshot();
  expect(wrapper.contains('Jogador 2')).toEqual(true);
});
