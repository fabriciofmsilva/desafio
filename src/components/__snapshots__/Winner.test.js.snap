// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Winner correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Winner
    gameover={false}
    players={
        Array [
            Object {
              "id": 1,
              "life": 0,
              "name": "Jogador 1",
            },
            Object {
              "id": 2,
              "life": 100,
              "name": "Jogador 2",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "Jogador 2",
        " venceu!",
      ],
      "className": "Winner hidden",
    },
    "ref": null,
    "rendered": Array [
      "Jogador 2",
      " venceu!",
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Jogador 2",
          " venceu!",
        ],
        "className": "Winner hidden",
      },
      "ref": null,
      "rendered": Array [
        "Jogador 2",
        " venceu!",
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`should render Winner correctly when gameover is false 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Winner
    gameover={false}
    players={
        Array [
            Object {
              "id": 1,
              "life": 100,
              "name": "Jogador 1",
            },
            Object {
              "id": 2,
              "life": 100,
              "name": "Jogador 2",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "Jogador 1",
        " venceu!",
      ],
      "className": "Winner hidden",
    },
    "ref": null,
    "rendered": Array [
      "Jogador 1",
      " venceu!",
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Jogador 1",
          " venceu!",
        ],
        "className": "Winner hidden",
      },
      "ref": null,
      "rendered": Array [
        "Jogador 1",
        " venceu!",
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`should render Winner correctly when gameover is true 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Winner
    gameover={true}
    players={
        Array [
            Object {
              "id": 1,
              "life": 100,
              "name": "Jogador 1",
            },
            Object {
              "id": 2,
              "life": 100,
              "name": "Jogador 2",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "Jogador 1",
        " venceu!",
      ],
      "className": "Winner zoomIn",
    },
    "ref": null,
    "rendered": Array [
      "Jogador 1",
      " venceu!",
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Jogador 1",
          " venceu!",
        ],
        "className": "Winner zoomIn",
      },
      "ref": null,
      "rendered": Array [
        "Jogador 1",
        " venceu!",
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
